app/assets/javascripts/discourse/app/routes/discovery-filter.js

```diff
@@ -34,6 +37,17 @@ export default class extends DiscourseRoute {
     });
   }
 
+  // This is required because by default Ember router will serialize Array type query param with JSON.stringify which
+  // results in a query param like `tags=tag1,tag2` which is not what we want. By doing nothing here, the query param
+  // will be serialized as `tags[]=tag1&tags[]=tag2`.
+  serializeQueryParam(value, urlKey, type) {
+    if (type === "customArray") {
+      return value;
+    } else {
+      return super.serializeQueryParam(value, urlKey, type);
+    }
+  }
```

Comment:
@davidtaylorhq This is not ideal as we're overriding a private method but I'm wonder if you might know another way of serializing query params.